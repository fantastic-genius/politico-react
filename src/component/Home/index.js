import React from 'react';


const Home = () => (
  <div>
    <h1>Politico React Home</h1>
  </div>
);

export default Home;
