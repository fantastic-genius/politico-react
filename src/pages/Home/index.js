import React from 'react';
import { Link } from 'react-router-dom';
import NavBar from '@components/NavBar';
import Footer from '@components/Footer';
import Button from '@components/Button';

const Home = () => (
    <div className="container home-con" >
      <NavBar />
      <div className="main">
        <div className="home">
          <div className="overlay">
            <div className="home-item-con">
              <h1>POLITICO</h1>
              <article>
                  <p>
                      This is an application where users can vote for election. In addition you can apply as a candidate
                      under a party and also cast votes for those who have shown interest for an office
                  </p>
              </article>
              <div className="home-btn">
                  <Link to='/signup'>
                    <Button className="btn" value='Sign Up'/>
                  </Link>
                  <Link to='/login'>
                    <Button className="btn" value='Login'/>
                  </Link>
              </div>
            </div>
          </div> 
        </div>
      </div>
      <Footer />
    </div>
  )

export default Home;
