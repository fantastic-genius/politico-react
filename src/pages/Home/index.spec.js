import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom'
import Home from './index';

describe('<Home />', () => {
  describe('Rendering', () => {
    it('should render Home page correctly', () => {
      const component = shallow(
        <Router>
          <Home />
        </Router>
      );
      expect(component).toMatchSnapshot();
    });

    it('should render Home page correctly woth its children', () => {
        const component = mount(
          <Router>
            <Home />
          </Router>
        );
        expect(component).toMatchSnapshot();
        expect(component.find('button')).toHaveLength(4)
    });
  });

})
