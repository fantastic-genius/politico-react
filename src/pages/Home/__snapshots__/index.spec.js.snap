// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Rendering should render Home page correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Home />
</Router>
`;

exports[`<Home /> Rendering should render Home page correctly woth its children 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Home>
      <div
        className="container home-con"
      >
        <NavBar>
          <div>
            <header>
              <div
                className="app-name"
              >
                <Link
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <h2>
                      POLITICO
                    </h2>
                  </a>
                </Link>
              </div>
              <div
                className="sign-btn"
              >
                <span
                  className="home-btn"
                >
                  <Link
                    to="/signup"
                  >
                    <a
                      href="/signup"
                      onClick={[Function]}
                    >
                      <Button
                        className="btn"
                        value="Sign Up"
                      >
                        <button
                          className="btn"
                        >
                          Sign Up
                        </button>
                      </Button>
                    </a>
                  </Link>
                  <Link
                    to="login"
                  >
                    <a
                      href="/login"
                      onClick={[Function]}
                    >
                      <Button
                        className="btn"
                        value="Login"
                      >
                        <button
                          className="btn"
                        >
                          Login
                        </button>
                      </Button>
                    </a>
                  </Link>
                </span>
              </div>
            </header>
          </div>
        </NavBar>
        <div
          className="main"
        >
          <div
            className="home"
          >
            <div
              className="overlay"
            >
              <div
                className="home-item-con"
              >
                <h1>
                  POLITICO
                </h1>
                <article>
                  <p>
                    This is an application where users can vote for election. In addition you can apply as a candidate under a party and also cast votes for those who have shown interest for an office
                  </p>
                </article>
                <div
                  className="home-btn"
                >
                  <Link
                    to="/signup"
                  >
                    <a
                      href="/signup"
                      onClick={[Function]}
                    >
                      <Button
                        className="btn"
                        value="Sign Up"
                      >
                        <button
                          className="btn"
                        >
                          Sign Up
                        </button>
                      </Button>
                    </a>
                  </Link>
                  <button
                    className="btn"
                  >
                    <a
                      href="UI/login.html"
                    >
                      Login
                    </a>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <Footer>
          <footer>
            <span>
              Copyright@2019  @fantastic-genius (Andela Bootcamp Cycle41)
            </span>
          </footer>
        </Footer>
      </div>
    </Home>
  </Router>
</BrowserRouter>
`;
